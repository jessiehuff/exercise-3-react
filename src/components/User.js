import React, {Component} from 'react'; 
import {connect} from 'react-redux'; 

class User extends Component {

    render(){
        const user = this.props.user[0]
         
        return (
            <div className="user_info">
                <h2>{user.name}</h2>
                <ul>
                    <li><h4 className="specific_detail">Username: </h4> {user.username}</li> 
                    <li><h4 className="specific_detail">Email: </h4> {user.email}</li> 
                    <li><h4 className="specific_detail">Address: </h4> {user.address.street}, {user.address.suite}, {user.address.city}, {user.address.zipcode} 
                        <p className="geo">Geo: {user.address.geo.lat}, {user.address.geo.lng}</p></li> 
                    <li><h4 className="specific_detail">Phone: </h4> {user.phone}</li> 
                    <li><h4 className="specific_detail">Website: </h4> {user.website}</li>  
                    <li><h4 className="specific_detail">Company: </h4> {user.company.name}
                        <p className="co_detail">{user.company.catchPhrase}</p>
                        <p className="co_detail">{user.company.bs}</p> 
                    </li> 
                </ul> 
                <br></br>
                <br></br> 
                <a href="/"><button>Back to All Users</button></a> 
            </div> 
        )
    }
}

const mapStateToProps = (state, ownProps) => {  
     
    const allUsers = state.users
    const user = allUsers.filter(user => user.id === parseInt(ownProps.match.params.userId))

    return { user }
}

export default connect(mapStateToProps, null)(User); 