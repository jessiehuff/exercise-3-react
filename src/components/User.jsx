import React, {Component} from 'react'; 
import {connect} from 'react-redux'; 

class User extends Component {
   


    render(){
        const user = this.props.user
        return (
            <div className="user">
                {user.map((x) => {
                    return(
                        <div className="user_info">
                        <h2>{x.name}</h2>
                        <ul>
                            <li><h4 className="specific_detail">Username: </h4> {x.username}</li> 
                            <li><h4 className="specific_detail">Email: </h4> {x.email}</li> 
                            <li><h4 className="specific_detail">Address: </h4> {x.address.street}, {x.address.suite}, {x.address.city}, {x.address.zipcode} 
                                <p className="geo">Geo: {x.address.geo.lat}, {x.address.geo.lng}</p></li> 
                            <li><h4 className="specific_detail">Phone: </h4> {x.phone}</li> 
                            <li><h4 className="specific_detail">Website: </h4> {x.website}</li>  
                            <li><h4 className="specific_detail">Company: </h4> {x.company.name}
                                <p className="co_detail">{x.company.catchPhrase}</p>
                                <p className="co_detail">{x.company.bs}</p> 
                            </li> 
                        </ul>   
                        <br></br>
                        <br></br> 
                        <a href="/"><button>Back to All Users</button></a>
                        </div> 
                    )
                })}
            </div> 
        )
    }
}

function mapStateToProps(state, ownProps){   
    const allUsers = state.users
    const user = allUsers.filter(user => user.id === parseInt(ownProps.match.params.userId))

    if (user) {
        return { user }
    }
    else {
    return { user: []}
    }
}

export default connect(mapStateToProps)(User); 