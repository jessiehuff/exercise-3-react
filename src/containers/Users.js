import React, {Component} from 'react'; 
import PropTypes from 'prop-types'; 
import {bindActionCreators} from 'redux'; 
import { connect } from 'react-redux'; 
import * as actionCreators from '../actions/index.js'; 


class Users extends Component {

    componentDidMount(){
        this.props.actions.fetchUsers(); 
    }

    render(){
        const allUsers = Object.keys(this.props).map(i => this.props[i])
        console.log(allUsers) 
        return(
            <p>Hello</p> 
            
        ) 
    } 
}

Users.propTypes = {
    users: PropTypes.array,
    actions: PropTypes.object
}; 

function mapStateToProps(state){
    return {
        ...state.users
    };
}

function mapDispatchToProps(dispatch){
    return {actions: bindActionCreators(actionCreators, dispatch)};
}

export default connect(mapStateToProps, mapDispatchToProps)(Users); 