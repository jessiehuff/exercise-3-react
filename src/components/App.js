import React, {Component} from 'react';
import { BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import { connect } from 'react-redux'; 
import { bindActionCreators } from 'redux';
import * as actionCreators from '../actions/index';
import Users from './Users';
import User from './User'; 

class App extends Component {    

    componentDidMount(){
        if (this.props.users.length === 0) {
            this.props.actions.fetchUsers(); 
        } 
    }

    render(){ 
        return (
            <Router>
                <Switch>
                    <Route exact path="/" component={Users} /> 
                    <Route exact path="/users/:userId" component={User} /> 
                </Switch>
            </Router>
        );
    }
}

function mapStateToProps(state){   
    return {
        users: state.users
    };
}

function mapDispatchToProps(dispatch) {
    return { actions: bindActionCreators(actionCreators, dispatch) };
  }

export default connect(mapStateToProps, mapDispatchToProps)(App); 