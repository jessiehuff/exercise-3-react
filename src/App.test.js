import React from 'react';
import Enzyme, {shallow} from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16'; 
import {shallowToJson} from 'enzyme-to-json'; 

import asyncFetch from './async-fetch.js'; 
import App from '../src/components/App';
import Users from '../src/components/Users'; 
import User from '../src/components/User'; 

const fetchMock = require('fetch-mock'); 
Enzyme.configure({ adapter: new Adapter() })


test('should render App correctly', () => {
  const output = shallow(
    <App url="/"/> 
  );
  expect(shallowToJson(output)).toMatchSnapshot(); 
});


test('should render Users component', () => {
  const output = shallow(
    <Users url="/" />
  );
  expect(shallowToJson(output)).toMatchSnapshot(); 
});

test('should render User component', () => {
  const output = shallow(
    <User url="/users/:userId"/>
  );
  expect(shallowToJson(output)).toMatchSnapshot(); 
});


//Testing data source
test('can fetch user data', async () => {
  fetchMock.get('http://jsonplaceholder.typicode.com/users', {name: "Leanne Graham"});

  const response = await asyncFetch('http://jsonplaceholder.typicode.com/users');
  const result = await response.json();

  expect(result.name).toEqual("Leanne Graham");
});