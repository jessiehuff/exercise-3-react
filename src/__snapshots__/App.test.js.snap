// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`App should render user detail page correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`App should render users page correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render App correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render URL correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render User component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render Users component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render user detail URL correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render user detail page correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render user detail page correctly 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render users URL correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render users page correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
